docs/source/en/model_doc/qwen2_audio.md

```diff
@@ -0,0 +1,188 @@
+
+
+## Usage tips
+
+`Qwen2-Audio-7B` and `Qwen2-Audio-7B-Instruct` can be found on the [Huggingface Hub](https://huggingface.co/Qwen)
+
+In the following, we demonstrate how to use `Qwen2-Audio-7B-Instrucct` for the inference, supporting both voice chat and audio analysis modes. Note that we have used the ChatML format for dialog, in this demo we show how to leverage `apply_chat_template` for this purpose.
+
+### Voice Chat Inference
+In the voice chat mode, users can freely engage in voice interactions with Qwen2-Audio without text input:
+```python
+from transformers import Qwen2AudioForConditionalGeneration, AutoProcessor
+from transformers.pipelines.audio_utils import ffmpeg_read
+import requests
+
+processor = AutoProcessor.from_pretrained("Qwen/Qwen2-Audio-7B-Instruct")
+model = Qwen2AudioForConditionalGeneration.from_pretrained("Qwen/Qwen2-Audio-7B-Instruct", device_map="auto")
+
+conversation = [
+            {"role": "user", "content": [
+                {"type": "audio", "audio_url": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/guess_age_gender.wav"},
+            ]},
+            {"role": "assistant", "content": "Yes, the speaker is female and in her twenties."},
+            {"role": "user", "content": [
+                {"type": "audio", "audio_url": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen2-Audio/audio/translate_to_chinese.wav"},
+            ]},
+        ]
+text = processor.apply_chat_template(conversation, add_generation_prompt=True, tokenize=False, add_audio_id = True)
+audios = []
+for message in conversation:
+    if isinstance(message["content"], list):
+        for ele in message["content"]:
+            if ele["type"] == "audio":
+                audios.append(ffmpeg_read(requests.get(ele['audio_url']).content, sampling_rate=processor.feature_extractor.sampling_rate))
+
+inputs = processor(text=text, audios=audios, return_tensors="pt", padding=True)
+inputs.input_ids = inputs.input_ids.to("cuda")
+
+generate_ids = model.generate(**inputs, max_length=256)
+generate_ids = generate_ids[:, inputs.input_ids.size(1):]
```

Comment:
As discussed in the modeling test, `generate_ids` should be made of entirely newly generated tokens, and you shouldn't have to pass the input ids length to get the generated tokens.

Reply:
Thanks, I haven't caught that! 

cc @amyeroberts and @ArthurZucker , is this expected? Thanks for your insights

Reply:
This has always been the behavior in `generate` for decoder-only models -- the model continues the input, so it makes sense for the whole thing to stay on a single tensor. 

This is different from the case with encoder-decoder models, where the generated content starts anew (because the input is NOT fed to the decoder). If you pass a started decoder sequence to an encoder-decoder model through `decoder_input_ids`, you'll also see this sequence in the output 🤗 

Because it is simple to crop and `generate` is meant to be a lower-level API, I'd be biased against adding a flag to output exclusively new tokens, unless there is a big demand for this functionality.